specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json

```diff
@@ -523,7 +526,32 @@
           "type": "integer",
           "minimum": 1,
           "maximum": 102400,
-          "description": "The maximum size of the share, in gigabytes. Must be greater than 0, and less than or equal to 5TB (5120). For Large File Shares, the maximum size is 102400."
+          "description": "The maximum size of the share, in gigabytes. Must be greater than 0, and less than or equal to 5TB (5120). For Large File Shares, the maximum size is 102400."        },
+        "enabledProtocols": {
+          "type": "string",
+          "enum": [
+            "SMB",
+            "NFS"
+          ],
+          "x-ms-enum": {
+            "name": "EnabledProtocols",
+            "modelAsString": true
+          },
+          "description": "Immutable property for file shares protocol.",
+          "x-ms-mutability": ["create", "read"]
+        },
+        "rootSquash": {
+          "type": "string",
+          "enum": [
+            "NoRootSquash",
+            "RootSquash",
+            "AllSquash"
+          ],
+          "x-ms-enum": {
+            "name": "RootSquashType",
+            "modelAsString": true
+          },
+          "description": "Reduction of the access rights for the remote superuser."
```

Comment:
Please add more description for What's the effect when specify each of the 3 values. What's the default value.